freqtrade/leverage/liquidation_price.py

```diff
@@ -0,0 +1,229 @@
+from typing import Optional
+
+from freqtrade.enums import Collateral, TradingMode, MarginMode
+from freqtrade.exceptions import OperationalException
+
+
+def liquidation_price(
+    exchange_name: str,
+    open_rate: float,
+    is_short: bool,
+    leverage: float,
+    trading_mode: TradingMode,
+    collateral: Optional[Collateral],
+    margin_mode: Optional[MarginMode]
+) -> Optional[float]:
+    if trading_mode == TradingMode.SPOT:
+        return None
+
+    if not collateral:
+        raise OperationalException(
+            "Parameter collateral is required by liquidation_price when trading_mode is "
+            f"{trading_mode}"
+        )
+
+    if exchange_name.lower() == "binance":
+        if not margin_mode:
+            raise OperationalException(
+                f"Parameter margin_mode is required by liquidation_price when exchange is {trading_mode}")
+
+        return binance(open_rate, is_short, leverage, margin_mode, trading_mode, collateral)
+    elif exchange_name.lower() == "kraken":
+        return kraken(open_rate, is_short, leverage, trading_mode, collateral)
+    elif exchange_name.lower() == "ftx":
+        return ftx(open_rate, is_short, leverage, trading_mode, collateral)
+    raise OperationalException(
+        f"liquidation_price is not implemented for {exchange_name}"
+    )
+
+
+def exception(
+    exchange: str,
+    trading_mode: TradingMode,
+    collateral: Collateral,
+    margin_mode: Optional[MarginMode] = None
+):
+    """
+        Raises an exception if exchange used doesn't support desired leverage mode
+        :param exchange: Name of the exchange
+        :param margin_mode: one-way or hedge
+        :param trading_mode: spot, margin, futures
+        :param collateral: cross, isolated
+    """
+    if not margin_mode:
+        raise OperationalException(
+            f"{exchange} does not support {collateral.value} {trading_mode.value} trading ")
+
+    raise OperationalException(
+        f"{exchange} does not support {collateral.value} {margin_mode.value} Mode {trading_mode.value} trading ")
+
+
+def binance(
+    open_rate: float,
+    is_short: bool,
+    leverage: float,
+    margin_mode: MarginMode,
+    trading_mode: TradingMode,
+    collateral: Collateral,
+    wallet_balance: float,
+    maintenance_margin_ex_1: float,
+    unrealized_pnl_ex_1: float,
+    maintenance_amount_both: float,
+    maintenance_amount_long: float,
+    maintenance_amount_short: float,
+    position_1_both: float,
+    entry_price_1_both: float,
+    position_1_long: float,
+    entry_price_1_long: float,
+    position_1_short: float,
+    entry_price_1_short: float,
+    maintenance_margin_rate_both: float,
+    maintenance_margin_rate_long: float,
+    maintenance_margin_rate_short: float,
```

Comment:
I'm trying to think if there's a better way method to organize parameters, it looks like the "both" parameters are never used at the same time as the long and short parameters, so maybe if there was just a maintenance_margin param that was a tuple, with only 1 value for one-way. Having the other exchange formulas will also help this, so that you can line up there parameters in the parent function.

It would be best not to do too much more for hedge mode right now though, you can try to organize it so that you don't have any conflicts when you try to revert back to using hedge mode, but definately don't write any tests for hedge mode or anything, like right now we don't need to worry about how to organize the parameters, but when it's done, and we also have the kraken and ftx formulas, then we can decide how to include hedge mode so that everything aligns with the other exchanges as well

Reply:
yea I think we can think about organising later when other exchange formulas are there. Right now I want to get it to a point where we can start testing this in dry mode

Reply:
I think now I will remove the hedge mode and move it to another branch on my fork and keep it. Let's get the tests done for this so we are ready to merge this.

Reply:
ok, I'll need an example of argument and return values to write the one test example, and then you'll be able to write the rest

Reply:
Tests added at https://github.com/arunavo4/freqtrade/pull/5576/commits/98d2d2f485f9605b5aa8eb7dd674b3bb3150f588